core/slice.go

```diff
@@ -272,6 +284,71 @@ func (sl *Slice) CollectSubRollups(b *types.Block) ([]types.Transactions, error)
 	return subRollups, nil
 }
 
+// CollectNewlyConfirmedEtxs collects all newly confirmed ETXs since the last coincident with the given location
+func (sl *Slice) CollectNewlyConfirmedEtxs(block *types.Block, location common.Location) (types.Transactions, types.Transactions, error) {
+	nodeCtx := common.NodeLocation.Context()
+	// Collect rollup of ETXs from the subordinate node's manifest
+	if nodeCtx == common.PRIME_CTX {
+		fmt.Printf("collecting pending ETXs @ block %s\n", block.Hash())
+	}
+	referencableEtxs := types.Transactions{}
+	subRollup := types.Transactions{}
+	if nodeCtx < common.ZONE_CTX {
+		subRollups, err := sl.CollectSubRollups(block)
+		if err != nil {
+			return nil, nil, err
+		}
+		for _, etxs := range subRollups {
+			referencableEtxs = append(referencableEtxs, etxs...)
+		}
+		referencableEtxs = append(referencableEtxs, block.ExtTransactions()...) // Include ETXs emitted in this block
+		subRollup = subRollups[nodeCtx+1]
+	}
+
+	// Filter for ETXs destined to this slice
+	newInboundEtxs := referencableEtxs.FilterToSlice(location, nodeCtx)
+
+	// Filter this list to exclude any ETX for which we are not the crossing
+	// context node. Such ETXs cannot be used by our subordinate for one of the
+	// following reasons:
+	// * if we are prime, but common dom was a region node, than the given ETX has
+	//   already been confirmed and passed down from the region node
+	// * if we are region, but the common dom is prime, then the destination is
+	//   not in one of our sub chains
+	//
+	// Note: here "common dom" refers to the highes context chain which exists in
+	// both the origin & destination. See the definition of the `CommonDom()`
+	// method for more explanation.
+	newlyConfirmedEtxs := newInboundEtxs.FilterConfirmationCtx(nodeCtx)
+
+	// Terminate the search if we reached genesis
+	if block.NumberU64() == 0 {
+		if block.Hash() != sl.config.GenesisHash {
+			return nil, nil, fmt.Errorf("terminated search on bad genesis, block0 hash: %s", block.Hash().String())
+		} else {
+			return newlyConfirmedEtxs, subRollup, nil
+		}
+	}
+	ancHash := block.ParentHash()
+	ancNum := block.NumberU64() - 1
+	ancestor := sl.hc.GetBlock(ancHash, ancNum)
+	if ancestor == nil {
+		return nil, nil, fmt.Errorf("unable to find ancestor, hash: %s", ancHash.String())
+	}
+	// Terminate the search when we find a block produced by the given location
+	if ancestor.Location().Equal(location) {
+		return newlyConfirmedEtxs, subRollup, nil
+	}
+
+	// Otherwise recursively process the ancestor and collect its newly confirmed ETXs too
+	ancEtxs, _, err := sl.CollectNewlyConfirmedEtxs(ancestor, location)
```

Comment:
This seems like a highly inefficient process. We should ideally store the ETXs with the block after they are collected so that we can simply look up and update from the previous parent calculation.  This obviously would need to be a cache.  Can we please make an issue to add this short circuit.

Reply:
just go with it.

Reply:
NVM